File: devices-api-java-mvc/src/main/java/com/lucas/devicesapijavamvc/api/dto/DeviceRequest.java
package com.lucas.devicesapijavamvc.api.dto;

import com.lucas.devicesapijavamvc.domain.DeviceState;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

public record DeviceRequest(
        @NotBlank String name,
        @NotBlank String brand,
        @NotNull DeviceState state
) {}

File: devices-api-kotlin-reactive/src/main/kotlin/com/lucas/devicesapikotlinreactive/infrastructure/api/dto/DeviceRequest.kt
package com.lucas.devicesapikotlinreactive.infrastructure.api.dto

import com.lucas.devicesapikotlinreactive.domain.model.DeviceState
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

data class DeviceRequest(
    @field:NotBlank
    val name: String,

    @field:NotBlank
    val brand: String,

    @field:NotNull
    val state: DeviceState
)
